packages/aws-cdk/lib/api/util/sdk.ts

```diff
@@ -389,6 +406,15 @@ async function hasEc2Credentials() {
   return instance;
 }
 
+async function setConfigVariable() {
+  const homeDir = process.env.HOME || process.env.USERPROFILE
+    || (process.env.HOMEPATH ? ((process.env.HOMEDRIVE || 'C:/') + process.env.HOMEPATH) : null) || os.homedir();
```

Comment:
Wouldn't `os.homedir()` just do all of this for you?

Reply:
Don't know. This is exactly what the SDK code does, I don't want to risk us going out of sync because there's some environment variable effect somewhere that we haven't considered.